ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/LocallySupportedEngineApiCapabilitiesProvider.java

```diff
@@ -32,46 +31,46 @@
 import tech.pegasys.teku.ethereum.executionclient.methods.EthGetBlockByHash;
 import tech.pegasys.teku.ethereum.executionclient.methods.EthGetBlockByNumber;
 import tech.pegasys.teku.spec.Spec;
-import tech.pegasys.teku.spec.SpecMilestone;
 
-public class MilestoneBasedExecutionJsonRpcMethodsProvider
-    implements ExecutionJsonRpcMethodsProvider {
+public class LocallySupportedEngineApiCapabilitiesProvider
+    implements EngineApiCapabilitiesProvider {
 
-  @SuppressWarnings("rawtypes")
-  private final List<EngineJsonRpcMethod> methods = new ArrayList<>();
+  private final Collection<EngineJsonRpcMethod<?>> supportedMethods = new HashSet<>();
 
-  public MilestoneBasedExecutionJsonRpcMethodsProvider(
+  public LocallySupportedEngineApiCapabilitiesProvider(
       final Spec spec, final ExecutionEngineClient executionEngineClient) {
-    if (spec.isMilestoneSupported(SpecMilestone.BELLATRIX)) {
-      methods.add(new EthGetBlockByHash(executionEngineClient));
-      methods.add(new EthGetBlockByNumber(executionEngineClient));
-      methods.add(new EngineNewPayloadV1(executionEngineClient));
-      methods.add(new EngineGetPayloadV1(executionEngineClient, spec));
-      methods.add(new EngineForkChoiceUpdatedV1(executionEngineClient));
-      methods.add(new EngineExchangeTransitionConfigurationV1(executionEngineClient));
-    }
+    // Eth1 methods
+    supportedMethods.add(new EthGetBlockByHash(executionEngineClient));
+    supportedMethods.add(new EthGetBlockByNumber(executionEngineClient));
 
-    if (spec.isMilestoneSupported(SpecMilestone.CAPELLA)) {
-      methods.add(new EngineNewPayloadV2(executionEngineClient));
-      methods.add(new EngineGetPayloadV2(executionEngineClient, spec));
-      methods.add(new EngineForkChoiceUpdatedV2(executionEngineClient, spec));
-    }
+    // New Payload
+    supportedMethods.add(new EngineNewPayloadV1(executionEngineClient));
+    supportedMethods.add(new EngineNewPayloadV2(executionEngineClient));
+    supportedMethods.add(new EngineNewPayloadV3(executionEngineClient));
 
-    if (spec.isMilestoneSupported(SpecMilestone.DENEB)) {
-      methods.add(new EngineNewPayloadV3(executionEngineClient));
-      methods.add(new EngineGetPayloadV3(executionEngineClient, spec));
-      methods.add(new EngineGetBlobsBundleV1(executionEngineClient, spec));
-    }
+    // Get Payload
+    supportedMethods.add(new EngineGetPayloadV1(executionEngineClient, spec));
+    supportedMethods.add(new EngineGetPayloadV2(executionEngineClient, spec));
+    supportedMethods.add(new EngineGetPayloadV3(executionEngineClient, spec));
+
+    // Fork Choice Updated
+    supportedMethods.add(new EngineForkChoiceUpdatedV1(executionEngineClient));
+    supportedMethods.add(new EngineForkChoiceUpdatedV2(executionEngineClient, spec));
+
+    // Get Blobs Bundle
```

Comment:
if we want these to add value rather than just be 'useless', we could add the actual EL fn name so when we search for the EL FN we get these come up...